File: android/src/main/java/com/comscore/flutter/streaming/utils/ParsedDate.java
package com.comscore.flutter.streaming.utils;

import static com.comscore.flutter.Args.getArgValue;

public class ParsedDate {
    public final Integer year, month, day;
    public ParsedDate (Object data) {
        if (data == null) {
            year = null;
            month = null;
            day = null;
            return;
        }
        year = getArgValue(data, "year");
        month = getArgValue(data, "month");
        day = getArgValue(data, "day");
    }

    public boolean isValid() {
        return year != null && month != null && day != null;
    }
}

File: ios/Classes/streaming/utils/ParsedDate.m
#import "ParsedDate.h"

@implementation ParsedDate;


- (ParsedDate*)initWithData:(id) data {
    self = [super init];
    if(self) {
        if (data == [NSNull null] || data == nil) {
            self.year = -1;
            self.month = -1;
            self.day = -1;
            self.isValid = NO;
        } else {
            self.year = data[@"year"] != [NSNull null] ? [data[@"year"] intValue] : -1;
            self.month = data[@"month"] != [NSNull null] ? [data[@"month"] intValue] : -1;
            self.day = data[@"day"] != [NSNull null] ? [data[@"day"] intValue] : -1;
            self.isValid = !(self.year < 0 || self.month < 0 || self.day < 0);
        }
    }
    return self;
}

@end
